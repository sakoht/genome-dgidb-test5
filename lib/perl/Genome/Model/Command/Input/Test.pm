
###########################################################################

package Genome::Model::Command::Input::Test;

use strict;
use warnings;

use base 'Test::Class';

use Data::Dumper 'Dumper';
use Test::More;

sub test_class {
    return 'Genome::Model::Command::Input';
}



###########################################################################

package Genome::Model::Command::Input::TestBase;

use strict;
use warnings;

use base 'Genome::Utility::TestCommandBase';

use Carp 'confess';
use Data::Dumper 'Dumper';
use Test::More;
require Genome::Model::Test;
require File::Temp;
require File::Path;

sub valid_param_sets {
    return grep { $_->{model_identifier} = $_[0]->_model->id } $_[0]->_valid_param_sets;
}

sub invalid_param_sets {
    return map { $_->{model_identifier} = $_[0]->_model->id } $_[0]->_valid_param_sets;
}

sub _model {
    my $self = shift;

    unless ( $self->{_model} ) {
        $self->{_model} = Genome::Model::Test->create_mock_model(
            type_name => 'tester',
            instrument_data_count => 0,
        )
            or confess "Can't create mock tester model.";
    }

    return $self->{_model};
}

sub _build {
    my $self = shift;

    return ($self->_model->builds)[0];
}



sub startup : Tests(startup => no_plan) {
    my $self = shift;
    
    no warnings;
    *Genome::Model::Command::get_model_type_names = sub{ 
        return (qw/ tester /); 
    };
    use warnings;
    is_deeply(
        [ Genome::Model::Command->get_model_type_names ],
        [qw/ tester /],
        'Set tester type name.',
    );

    $self->_startup or confess "Failed startup.";
    
    return 1;
}

sub _startup {
    return 1;
}

###########################################################################

package Genome::Model::Command::Input::Add::Test;

use strict;
use warnings;

use base 'Genome::Model::Command::Input::TestBase';

use Data::Dumper 'Dumper';
use Test::More;

sub test_class {
    return 'Genome::Model::Command::Input::Add';
}

sub _valid_param_sets {
    return (
        { # add multiple 'value_ids' 
            name => 'friends',
            ids => 'Watson,Crick',
            before_execute => '_before_execute',
            after_execute => '_after_execute',
        },
        { # add single 'value'
            name => 'inst_data',
            ids => '2sep09.934pmaa1',
            before_execute => '_before_execute',
            after_execute => '_after_execute',
        },
    );
}

sub invalid_param_sets {
    return (
        { # try to add a not 'is_many' input
            name => 'coolness', 
            ids => 'none',
        },
        { # try to add Crick again
            name => 'friends',
            ids => 'Crick',
        },
        { # try to add id that don't exist
            name => 'inst_data',
            ids => '2sep09.934noexist',
        },
    );
}

sub _before_execute {
    my ($self, $obj, $param_set) = @_;

    my $name = $obj->name;
    my @ids = $self->_model->$name;
    ok(!@ids, "No $name found for model.");
    
    return 1;
}

sub _after_execute {
    my ($self, $obj, $param_set) = @_;

    my $name = $obj->name;
    my @ids = $self->_model->$name;
    ok(@ids, "Added $name to model.");
    
    return 1;
}

###########################################################################

package Genome::Model::Command::Input::Remove::Test;

use strict;
use warnings;

use base 'Genome::Model::Command::Input::TestBase';

use Carp 'confess';
use Data::Dumper 'Dumper';
use Genome::Utility::TestBase;
use Test::More;

sub test_class {
    return 'Genome::Model::Command::Input::Remove';
}

sub _valid_param_sets {
    return (
        { # remove multiple 'value_ids' 
            name => 'friends',
            ids => 'Watson,Crick',
        },
        { # remove single 'value'
            name => 'inst_data',
            ids => '2sep09.934pmaa1',
            abandon_builds => 1,
        },
    );
}

sub invalid_param_sets {
    return (
        {# try to remove a not 'is_many' input
            name => 'coolness', 
            ids => 'none',
        },
        { # try to remove input that is not linked to the model anymore
            name => 'inst_data',
            ids => '2sep09.934pmaa1',
        },
    );
}

sub friends {
    return 'Crick';
}

sub inst_data {
    return Genome::InstrumentData::Sanger->get('2sep09.934pmaa1')
}

sub _startup {
    my $self = shift;

    my $model = $self->_model;

    $model->add_friend('Crick');
    $model->add_friend('Watson');
    my @friends = $model->friends();
    is_deeply(\@friends, [qw/ Crick Watson /], 'Added friends to remove.');
    my $id = Genome::InstrumentData::Sanger->get('2sep09.934pmaa1')
        or confess "Can't get sange id 2sep09.934pmaa1";
    $model->add_inst_data($id);
    my @inst_data = $model->inst_data;
    is_deeply(\@inst_data, [ $id ], 'Added instr_data to remove');
    
    my $build = $self->_build;
    my $build_input = Genome::Utility::TestBase->create_mock_object(
        class => 'Genome::Model::Build::Input',
        name => 'instrument_data',
        value_id => '2sep09.934pmaa1',
        build_id => $build->id,
    );
    $build->mock('abandon', sub{ return $build->build_event->event_status('Abandoned'); });

    return 1;
}

sub _post_execute {
    my ($self, $obj) = @_;

    my $name = $obj->name;
    my @ids = $self->_model->$name;
    #print Dumper(\@ids);
    ok(!@ids, "Removed $name from model.");
    if ( $obj->abandon_builds ) {
        is($self->_build->status, 'Abandoned', 'Successfully abandoned build');
    }

    return 1;
}

###########################################################################

package Genome::Model::Command::Input::List::Test;

use strict;
use warnings;

use base 'Genome::Model::Command::Input::TestBase';

use Data::Dumper 'Dumper';
use Test::More;

sub test_class {
    return 'Genome::Model::Command::Input::List';
}

sub valid_param_sets {
    return (
        {
            filter => 'model_id='.$_[0]->_model->id,
        },
    );
}

sub invalid_param_sets {
}

sub _startup {
    my $self = shift;

    $self->_model->coolness('low');
    is($self->_model->coolness, 'low', 'Set coolness (input)');

    return 1;
}

###########################################################################

package Genome::Model::Command::Input::Names::Test;

use strict;
use warnings;

use base 'Genome::Utility::TestCommandBase';

use Data::Dumper 'Dumper';
use Test::More;

sub test_class {
    return 'Genome::Model::Command::Input::Names';
}

sub valid_param_sets {
    return (
        {
            type_name => 'tester',
        },
    );
}

sub invalid_param_sets {
    return (
        {
            type_name => 'none',
        },
    );
}

sub startup : Tests(startup) {
    my $self = shift;
    
    no warnings 'once';
    no warnings 'redefine';
    *Genome::Model::Command::get_model_type_names = sub{ 
        return (qw/ tester /); 
    };
    
    return 1;
}

###########################################################################

package Genome::Model::Command::Input::Update::Test;

use strict;
use warnings;

use base 'Genome::Model::Command::Input::TestBase';

use Carp 'confess';
use Data::Dumper 'Dumper';
use Test::More;

sub test_class {
    return 'Genome::Model::Command::Input::Update';
}

sub _valid_param_sets {
    return (
        { # remove single 'value'
            name => 'coolness',
            value => 'moderate',
        },
        { # remove single 'value'
            name => 'coolness',
            value => 'UNDEF',
        },
    );
}

sub invalid_param_sets {
    return (
        { # try to update an 'is_many' input
            name => 'friends',
            value => 'Watson',
        },
    );
}

###########################################################################

1;

=pod

=head1 Tests

=head1 Disclaimer

 Copyright (C) 2009 Washington University Genome Sequencing Center

 This script is distributed in the hope that it will be useful,
 but WITHOUT ANY WARRANTY or the implied warranty of MERCHANTABILITY
 or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public
 License for more details.

=head1 Author(s)

 Eddie Belter <ebelter@watson.wustl.edu>

=cut

#$HeadURL$
#$Id$

