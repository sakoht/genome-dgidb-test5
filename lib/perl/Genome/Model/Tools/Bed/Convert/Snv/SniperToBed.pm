package Genome::Model::Tools::Bed::Convert::Snv::SniperToBed;

use strict;
use warnings;

use Genome;

class Genome::Model::Tools::Bed::Convert::Snv::SniperToBed {
    is => ['Genome::Model::Tools::Bed::Convert::Snv'],
};

sub help_brief {
    "Tools to convert sniper SNV format to BED.",
}

sub help_synopsis {
    my $self = shift;
    return <<"EOS"
  gmt bed convert snv sniper-to-bed --source snps_all_sequences --output snps_all_sequences.bed
EOS
}

sub help_detail {                           
    return <<EOS
    This is a small tool to take SNV calls in sniper format and convert them to a common BED format (using the first five columns).
EOS
}

sub process_source {
    my $self = shift;
    
    my $input_fh = $self->_input_fh;
    
    my $fn;
    while(my $line = <$input_fh>) {
        my @fields = split("\t", $line);
        if (!defined $fn) {
            if (@fields == 10) {
                $fn = \&process_10_col;
            } elsif (@fields == 26) {
                $fn = \&process_26_col;
            } else {
                die "Unexpected number of columns in somatic sniper output, expected 10 or 26, got " . scalar(@fields);
            }
        }
        $fn->($self, @fields);
    }
    
    return 1;
}

sub process_10_col {
    my $self = shift;
    my ($chromosome, $position, $reference, $genotype, $somatic_score, $snv_qual, $tumor_rms_map_qual, $tumor_depth, $normal_depth) = @_;
    #position => 1-based position of the SNV
    #BED uses 0-based position of and after the event
    $self->write_bed_line($chromosome, $position - 1, $position, $reference, $genotype, $somatic_score, $tumor_depth);
}

sub process_26_col {
    my $self = shift;
    my ($chromosome,
        $position,
        $reference,
        $tumor_genotype,
        $normal_genotype,
        $somatic_score,
        $tumor_consensus_quality,
        $tumor_snv_quality,
        $tumor_rms_map_quality,
        $normal_consensus_quality,
        $normal_snv_quality,
        $normal_rms_map_quality,
        $tumor_depth,
        $normal_depth,
        $tumor_mean_base_qual_matching_reference,
        $tumor_mean_map_qual_matching_reference,
        $tumor_depth_matching_reference,
        $tumor_mean_base_qual_variant,
        $tumor_mean_map_qual_variant,
        $tumor_depth_variant,
        $normal_mean_base_qual_matching_reference,
        $normal_mean_map_qual_matching_reference,
        $normal_depth_matching_reference,
        $normal_mean_base_qual_variant,
        $normal_mean_map_qual_variant,
        $normal_depth_variant) = @_;


    #position => 1-based position of the SNV
    #BED uses 0-based position of and after the event
    $self->write_bed_line(
        $chromosome,
        $position - 1,
        $position,
        $reference,
        $tumor_genotype,
        $somatic_score,
        $tumor_depth
        );
}

1;
