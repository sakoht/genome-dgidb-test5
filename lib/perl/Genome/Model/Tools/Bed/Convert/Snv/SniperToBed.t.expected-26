11	17364979	17364980	C/Y	0	59
11	17365205	17365206	C/Y	0	60
11	35265358	35265359	C/Y	4	60
11	35265395	35265396	A/T	8	60
11	35265397	35265398	T/A	9	55
11	55015044	55015045	A/G	4	60
11	55015660	55015661	T/C	0	55
11	55459586	55459587	G/R	0	59
11	64115384	64115385	G/A	1	60
11	64115724	64115725	T/Y	27	55
11	64115725	64115726	C/Y	27	55
11	64115861	64115862	C/T	1	51
11	66367220	66367221	G/C	5	60
11	66367357	66367358	C/A	3	60
11	66367361	66367362	C/T	2	60
11	66367721	66367722	T/C	0	52
11	66367724	66367725	G/C	0	52
11	66367726	66367727	G/C	0	52
11	66367730	66367731	A/G	0	60
11	66368537	66368538	A/R	8	53
