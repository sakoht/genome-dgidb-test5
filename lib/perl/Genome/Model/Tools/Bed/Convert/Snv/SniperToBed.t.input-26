11	17364980	C	Y	Y	0	31	31	60	142	142	59	11	18	33	60	8	35	60	3	36	60	10	34	57	8
11	17365206	C	Y	Y	0	171	171	58	228	228	60	58	105	34	56	41	35	60	17	35	60	55	35	60	50
11	35265359	C	Y	C	4	3	5	60	33	0	60	2	2	36	60	1	38	60	1	38	60	2	0	0	0
11	35265396	A	T	A	8	30	30	60	36	0	60	1	3	0	0	0	37	60	1	33	60	3	0	0	0
11	35265398	T	A	T	9	30	30	60	39	0	55	1	4	0	0	0	38	60	1	32	54	4	0	0	0
11	55015045	A	G	G	4	36	36	60	30	30	60	3	1	0	0	0	36	60	3	0	0	0	38	60	1
11	55015661	T	C	C	0	138	138	55	132	132	55	37	35	0	0	0	35	54	37	0	0	0	32	54	35
11	55459587	G	R	R	0	228	228	56	145	145	59	40	21	35	54	24	36	57	16	36	58	13	36	60	8
11	64115385	G	A	A	1	57	57	53	36	36	60	10	3	0	0	0	34	51	10	0	0	0	33	60	3
11	64115725	T	Y	T	27	30	30	60	96	0	55	14	23	37	60	2	34	60	2	33	54	23	0	0	0
11	64115726	C	Y	C	27	30	30	60	105	0	55	14	26	38	60	2	34	60	2	29	54	26	0	0	0
11	64115862	C	T	T	1	57	57	57	34	57	51	10	11	0	0	0	32	55	10	2	0	1	28	52	10
11	66367221	G	C	C	5	30	30	60	30	30	60	1	1	0	0	0	38	60	1	0	0	0	38	60	1
11	66367358	C	A	C	3	29	29	29	36	0	60	1	3	0	0	0	35	29	1	36	60	3	0	0	0
11	66367362	C	T	C	2	29	29	29	36	0	60	1	3	0	0	0	37	29	1	35	60	3	0	0	0
11	66367722	T	C	T	0	0	0	47	28	0	52	2	3	2	60	1	2	29	1	38	60	2	0	0	0
11	66367725	G	C	G	0	0	0	47	36	0	52	2	3	2	60	1	2	29	1	26	50	3	0	0	0
11	66367727	G	C	G	0	0	0	47	36	0	52	2	3	2	60	1	2	29	1	26	50	3	0	0	0
11	66367731	A	G	A	0	4	4	60	33	0	60	1	2	0	0	0	2	60	1	38	60	2	0	0	0
11	66368538	A	R	A	8	10	10	60	105	0	53	11	26	33	60	9	38	60	2	36	52	26	0	0	0
