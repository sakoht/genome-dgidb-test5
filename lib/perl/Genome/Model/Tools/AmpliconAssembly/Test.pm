###########################################################################

package Genome::Model::Tools::AmpliconAssembly::TestBase;

use strict;
use warnings;

use base 'Genome::Utility::TestBase';

use Data::Dumper 'Dumper';
require Genome::Model::Test;
use Test::More;

sub params_for_test_class {
    return (
        directory => $_[0]->amplicon_assembly->directory, # this creates the aa
        $_[0]->_params_for_test_class,
    );
}

sub amplicon_assembly {
    my $self = shift;

    unless ( $self->{_amplicon_assembly} ) {
        $self->{_amplicon_assembly} = Genome::Model::Tools::AmpliconAssembly::Set->create(
            directory => $self->tmp_dir,
        );
    }

    return $self->{_amplicon_assembly};
}

sub amplicons {
    return $_[0]->{_object}->get_amplicons;
}

sub _build_dir {
    return $_[0]->base_test_dir.'/Genome-Model-AmpliconAssembly/build-10000';
}

sub _params_for_test_class { return; }
sub should_copy_traces { 1 }
sub should_copy_edit_dir { 1 }
sub _pre_execute { 1 }

sub test01_copy_data : Tests {
    my $self = shift;

    if ( $self->should_copy_traces ) {
        ok( 
            Genome::Model::Test->copy_test_dir(
                $self->_build_dir.'/chromat_dir',
                $self->tmp_dir.'/chromat_dir',
            ),
            "Copy traces"
        ) or die;
    }

    if ( $self->should_copy_edit_dir ) {
        ok(
            Genome::Model::Test->copy_test_dir(
                $self->_build_dir.'/edit_dir',
                $self->tmp_dir.'/edit_dir',
            ),
            "Copy edit_dir"
        ) or die;
    }

    return 1;
}

sub test02_execute : Test(2) {
    my $self = shift;

    ok($self->_pre_execute, 'Pre Execute')
        or die "Failed method _pre_execute\n";

    ok($self->{_object}->execute, "Execute");
    #print $self->{_object}->directory,"\n"; <STDIN>;

    return 1;
}

###########################################################################

package Genome::Model::Tools::AmpliconAssembly::AssembleTest;

use strict;
use warnings;

use base 'Genome::Model::Tools::AmpliconAssembly::TestBase';

use Data::Dumper 'Dumper';
use Test::More;

sub test_class {
    return 'Genome::Model::Tools::AmpliconAssembly::Assemble';
}

sub invalid_params_for_test_class {
    return (
        assembler_params => 'u-n-p-a-r-s able 00',
    );
}
sub _pre_execute {
    my $self = shift;

    my $amplicons = $self->amplicons;
    for my $amplicon ( @$amplicons ) {
        # remove ace files
        unlink $amplicon->ace_file;
    }

    my $cnt = grep { -s $_->ace_file } @$amplicons;
    die "Could not remove ace files\n" if $cnt;

    return 1;
}

sub test03_verify : Test(1) {
    my $self = shift;

    my $amplicons = $self->amplicons;
    my $ace_cnt = grep { -s $_->ace_file } @$amplicons;
    is($ace_cnt, @$amplicons, 'Verified - Created an acefile for each amplicon');
    
    return 1;
}

###########################################################################

package Genome::Model::Tools::AmpliconAssembly::ClassifyTest;

use strict;
use warnings;

use base 'Genome::Model::Tools::AmpliconAssembly::TestBase';

use Data::Dumper 'Dumper';
use Test::More;

sub test_class {
    return 'Genome::Model::Tools::AmpliconAssembly::Classify';
}

sub _pre_execute {
    my $self = shift;

    my $amplicons = $self->amplicons;
    for my $amplicon ( @$amplicons ) {
        my $class_file = $amplicon->classification_file;
        unlink $class_file if -e $class_file;
    }

    my $cnt = grep { -s $_->classification_file } @$amplicons;
    die "Could not remove classification files\n" if $cnt;

    return 1;
}

sub test03_verify : Test(1) {
    my $self = shift;

    my $amplicons = $self->amplicons;
    my $cnt = grep { -s $_->classification_file } @$amplicons;
    is($cnt, @$amplicons, 'Verified - Created classification for each amplicon');
    
    return 1;
}

###########################################################################

package Genome::Model::Tools::AmpliconAssembly::ContaminationScreenTest;

use strict;
use warnings;

use base 'Genome::Model::Tools::AmpliconAssembly::TestBase';

use Data::Dumper 'Dumper';
use Test::More;

sub test_class {
    return 'Genome::Model::Tools::AmpliconAssembly::ContaminationScreen';
}

sub _params_for_test_class {
    return (
        database => '/gsc/var/lib/reference/set/2809160070/blastdb/blast',
        remove_contaminants => 1,
    );
}

sub _pre_execute { 
    # copy in the contaminated reads
    # contaminated read is vjn39g07.g1
    my $self = shift;

    for my $subdir ( (qw/ chromat_dir edit_dir /) ){
        my $source = $self->dir."/$subdir";
        my $dest = $self->tmp_dir."/$subdir";
        my $dh = Genome::Sys->open_directory($source)
            or die;
        while ( my $file = $dh->read ) {
            next if $file =~ m#^\.#;
            File::Copy::copy("$source/$file", $dest)
                or die "Can't copy ($source/$file) to ($dest): $!\n";
        }
    }

    $self->{_pre_execute_amplicons} = [ map { $_->get_name } @{$self->amplicons} ];

    return 1;
}

sub test03_verify : Test(2) {
    my $self = shift;

    # screen file
    my $screen_file = $self->{_object}->screen_file;
    ok(-s $screen_file, "Wrote screen file: $screen_file");

    # make sure we removed the conaminated amp
    my %pre_execute_amps = map { $_ => 1 } @{$self->{_pre_execute_amplicons}};
    for my $amplicon ( @{$self->amplicons} ) {
        delete $pre_execute_amps{$amplicon->get_name};
    }
    is_deeply(
        [ keys %pre_execute_amps ],
        [qw/ uyj40e03 /],
        'Removed contaminated amplicon: uyj40e03'
    );
    
    return 1;
}

###########################################################################

package Genome::Model::Tools::AmpliconAssembly::CollateTest;

use strict;
use warnings;

use base 'Genome::Model::Tools::AmpliconAssembly::TestBase';

use Data::Dumper 'Dumper';
use Test::More;

sub test_class {
    return 'Genome::Model::Tools::AmpliconAssembly::Collate';
}

sub test03_verify : Test(2) {
    my $self = shift;

    my $collate = $self->{_object};
    my @types = $collate->amplicon_assembly->amplicon_fasta_types;
    my $fasta_cnt = grep { -s $collate->amplicon_assembly->fasta_file_for_type($_) } @types;
    is($fasta_cnt, @types, 'Verified - Created a fasta for each type');
    my $qual_cnt = grep { -s $collate->amplicon_assembly->qual_file_for_type($_) } @types;
    is($qual_cnt, @types, 'Verified - Created a qual for each type');
    
    return 1;
}

###########################################################################

package Genome::Model::Tools::AmpliconAssembly::CopyFromBuild::Test;

use strict;
use warnings;

use base 'Genome::Model::Tools::AmpliconAssembly::TestBase';

use Genome::Model::Test;
use Data::Dumper 'Dumper';
use Test::More;

sub test_class {
    return 'Genome::Model::Tools::AmpliconAssembly::CopyFromBuild';
}

sub params_for_test_class {
    return (
        directory => $_[0]->tmp_dir,
        build_id => $_[0]->_model->last_complete_build_id,
        exclude_contaminated_amplicons => 1,
        copy_reads_only => 1,
    );
}

sub _model {
    my $self = shift;

    unless ( $self->{_model} ) {
        $self->{_model} = Genome::Model::Test->create_mock_model(
            type_name => 'amplicon assembly',
            use_mock_dir => 1,
        )
            or die "Can't create mock aa model";
    }

    return $self->{_model};
}

sub should_copy_traces { 0 }
sub should_copy_edit_dir { 0 }

sub _pre_execute {
    $File::Copy::Recursive::KeepMode = 0;
    return 1;
}

sub test03_verify : Test(4) {
    my $self = shift;

    my $amplicon_assembly = Genome::Model::Tools::AmpliconAssembly::Set->get(directory => $self->tmp_dir);
    ok($amplicon_assembly, 'Created amplicon assembly');
    ok($amplicon_assembly->exclude_contaminated_amplicons, 'Excluding contaminated amplicons');
    ok(scalar(glob($amplicon_assembly->chromat_dir.'/*')), 'Copied reads');
    ok(!scalar(glob($amplicon_assembly->edit_dir.'/*')), 'Did not copy edit dir');
    
    return 1;
}

###########################################################################

package Genome::Model::Tools::AmpliconAssembly::Create::Test;

use strict;
use warnings;

use base 'Genome::Model::Tools::AmpliconAssembly::TestBase';

use Data::Dumper 'Dumper';
use Test::More;

sub test_class {
    return 'Genome::Model::Tools::AmpliconAssembly::Create';
}

sub params_for_test_class {
    return (
        directory => $_[0]->tmp_dir,
    );
}

sub should_copy_traces { 0 }
sub should_copy_edit_dir { 0 }

sub _pre_execute { 
    my $self = shift;

    die if -e $self->tmp_dir.'/properties.stor';
    
    return 1;
}

sub test03_verify : Test(1) {
    my $self = shift;

    ok(-e $self->tmp_dir.'/properties.stor', 'Created amplicon assembly');

    return 1;
}

###########################################################################

package Genome::Model::Tools::AmpliconAssembly::OrientTest;

use strict;
use warnings;

use base 'Genome::Model::Tools::AmpliconAssembly::TestBase';

use Data::Dumper 'Dumper';
use Test::More;

sub test_class {
    return 'Genome::Model::Tools::AmpliconAssembly::Orient';
}

sub _pre_execute {
    my $self = shift;

    my $amplicons = $self->amplicons;
    for my $amplicon ( @$amplicons ) {
        my $ori_fasta = $amplicon->oriented_fasta_file;
        unlink $ori_fasta if -e $ori_fasta;
        my $ori_qual = $amplicon->oriented_qual_file;
        unlink $ori_qual if -e $ori_qual;
    }

    my $cnt = grep { -s $_->oriented_fasta_file } @$amplicons;
    die "Did not remove oriented fastas\n" if $cnt;

    return 1;
}

sub test03_verify : Test(2) {
    my $self = shift;

    my $amplicons = $self->amplicons;
    my $fasta_cnt = grep { -s $_->oriented_fasta_file } @$amplicons;
    is($fasta_cnt, @$amplicons, 'Verified - Created oriented fasta for each amplicon');
    my $qual_cnt = grep { -s $_->oriented_qual_file } @$amplicons;
    is($qual_cnt, @$amplicons, 'Verified - Created oriented qual for each amplicon');
    
    return 1;
}

###########################################################################

package Genome::Model::Tools::AmpliconAssembly::PrepareData::Test;

use strict;
use warnings;

use base 'Genome::Model::Tools::AmpliconAssembly::TestBase';

use Data::Dumper 'Dumper';
use Test::More;

sub test_class {
    return 'Genome::Model::Tools::AmpliconAssembly::PrepareData';
}

sub should_copy_edit_dir { return; }

sub test03_verify : Test(1) {
    my $self = shift;

    my $amplicons = $self->amplicons;
    my $fasta_cnt = grep { -s $_->fasta_file } @$amplicons;
    is($fasta_cnt, @$amplicons, 'Verified - Created fasta for each amplicon');
    
    return 1;
}

###########################################################################

package Genome::Model::Tools::AmpliconAssembly::TrimAndScreenTest;

use strict;
use warnings;

use base 'Genome::Model::Tools::AmpliconAssembly::TestBase';

require File::Copy;
require File::Compare;
use Data::Dumper 'Dumper';
use Test::More;

sub test_class {
    return 'Genome::Model::Tools::AmpliconAssembly::TrimAndScreen';
}

sub _params_for_test_class {
    return (
        trimmer_and_screener => 'trim3_and_crossmatch',
        #trimmer_and_screener_params => '-;',
    );
}

sub invalid_params_for_test_class {
    return (
        trimmer_and_screener => 'consed',
        trimmer_and_screener_params => '-;',
    );
}

sub _pre_execute {
    my $self = shift;

    my $amplicons = $self->amplicons;
    for my $amplicon ( @$amplicons ) {
        my $check_file = sprintf(
            '%s/%s.check.fasta',
            $amplicon->directory,
            $amplicon->name,
        );
        File::Copy::copy($amplicon->reads_fasta_file, $check_file)
            or die "Can't copy ".$amplicon->reads_fasta_file." to ".$check_file;
        unlink $amplicon->fasta_file;
        unlink $amplicon->qual_file;
        File::Copy::move($amplicon->reads_fasta_file, $amplicon->fasta_file)
            or die "Can't copy ".$amplicon->reads_fasta_file." to ".$amplicon->fasta_file;
        File::Copy::move($amplicon->reads_qual_file, $amplicon->qual_file)
            or die "Can't copy ".$amplicon->reads_fasta_file." to ".$amplicon->fasta_file;
    }

    return 1;
}

sub test03_verify : Test(1) {
    my $self = shift;

    my $amplicons = $self->amplicons;
    my $compare_cnt = 0;
    for my $amplicon ( @$amplicons ) {
        my $pre_fasta = sprintf(
            '%s/%s.check.fasta',
            $amplicon->directory,
            $amplicon->name,
        );
        $compare_cnt++ if File::Compare::compare($amplicon->fasta_file, $pre_fasta);
    }
    is ($compare_cnt, scalar(@$amplicons), 'Trimmed and screened amplicons');

    return 1;
}

############################################################################

1;

