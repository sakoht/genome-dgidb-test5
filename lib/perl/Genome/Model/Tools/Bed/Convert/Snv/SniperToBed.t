#!/gsc/bin/perl

use strict;
use warnings;

use Test::More tests => 9;

use above 'Genome';

use_ok('Genome::Model::Tools::Bed::Convert::Snv::SniperToBed');

my $tmpdir = File::Temp::tempdir('Bed-Convert-Snv-SniperToBedXXXXX', DIR => '/gsc/var/cache/testsuite/running_testsuites/', CLEANUP => 1);

my $input_file = __FILE__ . '.input';
my $expected_file = __FILE__ . '.expected';

verify_output($input_file, "$tmpdir/10col.out", $expected_file);
verify_output("$input_file-26", "$tmpdir/26col.out", "$expected_file-26");

sub verify_output {
    my ($input_file, $output_file, $expected_file) = @_;
    my $command = Genome::Model::Tools::Bed::Convert::Snv::SniperToBed->create( source => $input_file, output => $output_file );
    ok($command, 'Command created');
    my $rv = $command->execute;
    ok($rv, 'Command completed successfully');
    ok(-s $output_file, "output file created");

    my $diff = Genome::Sys->diff_file_vs_file($output_file, $expected_file);
    ok(!$diff, 'output matched expected result')
        or diag("diff results:\n" . $diff);
}
