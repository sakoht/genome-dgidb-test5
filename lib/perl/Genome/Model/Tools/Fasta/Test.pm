package Genome::Model::Tools::Fasta::Trim::LucyTest;

use strict;
use warnings;

use base 'Genome::Utility::TestBase';

use File::Compare 'compare';
use Test::More;

sub lucy {
    return $_[0]->{_object};
}

sub test_class {
    return 'Genome::Model::Tools::Fasta::Trim::Lucy';
}

sub params_for_test_class {
    my $self = shift;
    return (
        vector_name => 'pCR4-TOPO',
        fasta_file => $self->fasta_file,
        output_fasta_file => $self->tmp_dir.'/lucy.fasta',
    );
}

sub required_attrs {
    return (qw/ vector_name fasta_file /);
}

sub fasta_file {
    return $_[0]->dir.'/H_KM-aab04h11.reads.fasta';
}

sub vector_fasta_file {
    return $_[0]->dir.'/pCR4-TOPO.fasta';
}

sub flanking_fasta_file {
    return $_[0]->dir.'/pCR4-TOPO.flanking.fasta';
}

sub test01_execute : Tests {
    my $self = shift;

    my $lucy = $self->lucy;

    # execute
    ok($lucy->execute, 'Executed lucy');

    # vector files
    diag( $self->vector_fasta_file.' => '.$lucy->vector_fasta_file );
    is( compare($self->vector_fasta_file, $lucy->vector_fasta_file), 0, 'Vector fasta file matches');
    diag( $self->flanking_fasta_file.' => '.$lucy->flanking_fasta_file );
    is( compare($self->flanking_fasta_file, $lucy->flanking_fasta_file), 0, 'Flanking fasta file matches');
    
    #print 'lucy => '.$lucy->_tmpdir,"\n".'fasta => '.$self->tmp_dir,"\n"; <STDIN>;

    return 1;
}

#####################################################################

package Genome::Model::Tools::Fasta::Trim::LucyReaderTest;

use strict;
use warnings;

use base 'Genome::Utility::TestBase';

use Data::Dumper 'Dumper';
use Test::More;

sub test_class {
    return 'Genome::Model::Tools::Fasta::Trim::LucyReader';
}

sub test_class_sub_dir {
    return 'Genome-Model-Tools-Fasta-Trim-Lucy';
}
    
sub params_for_test_class {
    return (
        input => $_[0]->lucy_file,
    );
}

sub lucy_file {
        return $_[0]->dir.'/reader.debug.txt',
}

sub lucies_file {
        return $_[0]->dir.'/reader.lucies.stor',
}

sub test01_read_and_verify : Tests {
    my $self = shift;

    my @lucies = $self->{_object}->all;
    #print Dumper(\@lucies);
    #unlink $self->lucies_file; $self->store_file(\@lucies, $self->lucies_file);
    my $stored_lucies = $self->retrieve_file($self->lucies_file);
    is_deeply(\@lucies, $stored_lucies, 'Generated and stored lucies match');
    
    return 1;
}

1;

=pod

=head1 Name

ModuleTemplate

=head1 Synopsis

=head1 Usage

=head1 Methods

=head2 

=over

=item I<Synopsis>

=item I<Arguments>

=item I<Returns>

=back

=head1 See Also

=head1 Disclaimer

Copyright (C) 2005 - 2008 Washington University Genome Sequencing Center

This module is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY or the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more details.

=head1 Author(s)

B<Eddie Belter> I<ebelter@watson.wustl.edu>

=cut

#$HeadURL$
#$Id$

