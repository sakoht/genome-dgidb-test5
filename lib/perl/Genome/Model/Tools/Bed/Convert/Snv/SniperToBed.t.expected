1	176	177	A/M	19
1	230	231	C/M	53
1	4461	4462	T/A	20
1	5052	5053	G/R	28
1	56344	56345	T/A	20
1	56378	56379	T/A	17
1	67003	67004	C/T	18
1	73755	73756	A/G	17
1	73761	73762	A/G	16
1	73765	73766	A/G	17
1	73769	73770	A/G	17
1	73773	73774	A/G	16
1	78602	78603	T/C	16
1	78628	78629	A/T	17
1	78629	78630	C/T	23
1	88859	88860	G/K	22
1	125879	125880	C/T	17
1	164795	164796	A/G	16
1	164797	164798	A/G	15
1	219764	219765	G/K	30
